src/modules/auth/auth.service.ts

```diff
@@ -61,6 +63,29 @@ export class AuthService {
     return { accessToken, refreshToken };
   }
 
+  async updatePassword(dto: UpdatePasswordDto, user: User) {
+    const isPasswordValid = await Hasher.verifyHash(
+      user.passwordHash,
+      dto.oldPassword,
+    );
+    if (!isPasswordValid) {
+      throw new BadRequestException('Invalid email or password');
+    }
+
+    const hashedPassword = await Hasher.hashValue(dto.newPassword);
+
```

Comment:
(Optional) You can also validate that the new password is not the same as the old password. Or maybe it's done in frontend.

Reply:
handled in frontend